flake.nix

```diff
@@ -122,6 +124,61 @@
         };
         formatter = pkgs.alejandra;
 
+        checks = let
+          nixos-lib = import "${nixpkgs}/nixos/lib" {};
+        in
+          if ! pkgs.lib.hasSuffix "linux" system
+          then {}
+          else let
+            cmdLineValidation = {
+              testConfig,
+              testScript,
+            }:
+              nixos-lib.runTest {
+                name = "tsnsrv-nixos";
+                hostPkgs = pkgs;
+
+                defaults.services.tsnsrv.enable = true;
+                defaults.services.tsnsrv.defaults.package = config.packages.tsnsrvCmdLineValidator;
+                defaults.services.tsnsrv.defaults.authKeyPath = "/dev/null";
```

Comment:
nixos has snakeoil certs available for testing https://github.com/NixOS/nixpkgs/tree/master/nixos/tests/common/acme/server

Reply:
Nice! I think I want to replace these tests with something real that uses headscale anyway - the stub commandline validator doesn't feel great, but it does the trick for now (and it doesn't really care about the existence of files). I'll keep the certs in mind when I redo the tests!